crates/sui-network/src/state_sync/mod.rs

```diff
@@ -138,6 +141,9 @@ struct PeerStateSyncInfo {
     on_same_chain_as_us: bool,
     /// Highest checkpoint sequence number we know of for this Peer.
     height: CheckpointSequenceNumber,
+    /// lowest available checkpoint watermark for this Peer.
+    /// This defaults to 0 for now.
```

Comment:
how does it work with defaulting to zero, once everyone has upgraded? will this ever cause us to request checkpoints from a peer that they might not have?

Reply:
Today we construct `PeerStateSyncInfo` only when for the first time we talk to this peer, by querying the genesis checkpoint summary. Later in every tick we poll peers to get their latest checkpoint availability (added in this PR). Theoretically it's possible to request a checkpoint before that info is updated. 

I think we should call `get_checkpoint_availability` to init `PeerStateSyncInfo` once this call is rolled out so we always have the correct data. And before that happens we need to bear with the small possibility of what you described.

Reply:
Maybe add a todo for this? But I guess it's a non issue until peers start pruning tx, is that right?